### DISCUSSION TITLE
Using Fullstaq Ruby with Passenger

### DISCUSSION THREAD

**Discussion by sbarrick** [fullstaq-labs/fullstaq-ruby-server-edition#108] [Q&A]
We're trying to use Fullstaq Ruby on an Ubuntu 20.04 EC2 instance with Passenger and Nginx , the latter two installed via the recommended process from [the Passenger website](https://www.phusionpassenger.com/docs/tutorials/deploy_to_production/installations/oss/aws/ruby/nginx/). 

The problem we're running into is that the passenger apt package depends on Ruby and apt doesn't recognize the Ruby installs from the Fullstaq repository as valid options to meet that dependency, so we end up with an Ubuntu provided Ruby version installed as well.

We've come "close" by configuring passenger options as :

```
passenger_root /usr/lib/ruby/vendor_ruby/phusion_passenger/locations.ini;
passenger_ruby /usr/lib/fullstaq-ruby/versions/2.7-jemalloc/bin/ruby;
```

But passenger still boots using the default OS provided Ruby rather than the specific version we install from the Fullstaq repo. A subset of the installed packages can be found here:

```
$ dpkg -l | grep ruby
ii  fullstaq-ruby-2.7-jemalloc         10-ubuntu-20.04                   amd64        Fullstaq Ruby 2.7-jemalloc
ii  fullstaq-ruby-common               1.0-0                             all          Fullstaq Ruby common dependencies
ii  ruby2.7                            2.7.0-5ubuntu1.7                  amd64        Interpreter of object-oriented scripting language Ruby
ii  ruby                               1:2.7+1                           amd64        Interpreter of object-oriented scripting language Ruby (default version)
```

Has anyone found a successfully configuration to make Fullstaq Ruby and Passenger work together?  Is there a different approach beyond installing the passenger gem and compiling nginx from source?  Our team was surprised to find little if any mention of using Fullstaq Ruby and Passenger together in our internet research.  Thanks!

**Comment by FooBarWidget** [comment]
Passenger relies on a Ruby package that provides /usr/bin/ruby because its CLI tools depend on that. Fullstaq Ruby does not provide /usr/bin/ruby and instead relies on either absolute paths or PATH-based activation in the shell, and thus cannot satisfy Passenger's Ruby dependency requirement. At least, not until #6 is implemented. So at the moment you'll have to accept that Passenger installs a non-Fullstaq Ruby dependency.
